### ISSUE TITLE
A new app generated by `blitz new` causes `Error: DYNAMIC_SERVER_USAGE`

### ISSUE THREAD

**Issue by exKAZUu** [blitz-js/blitz#4112]
### What is the problem?

A new app generated by `blitz new` (Blitz version: 2.0.0-beta.24) causes `Error: DYNAMIC_SERVER_USAGE`.
Reproduction repo with steps: https://github.com/exKAZUu/blitz-new-issue

### Paste all your error logs here:

```
Uncaught Error: DYNAMIC_SERVER_USAGE
    at updateDehydratedSuspenseComponent (react-dom.development.js?ac89:20662:1)
    at updateSuspenseComponent (react-dom.development.js?ac89:20362:1)
    at beginWork (react-dom.development.js?ac89:21624:1)
    at beginWork$1 (react-dom.development.js?ac89:27426:1)
    at performUnitOfWork (react-dom.development.js?ac89:26557:1)
    at workLoopSync (react-dom.development.js?ac89:26466:1)
    at renderRootSync (react-dom.development.js?ac89:26434:1)
    at performConcurrentWorkOnRoot (react-dom.development.js?ac89:25738:1)
    at workLoop (scheduler.development.js?c964:266:1)
    at flushWork (scheduler.development.js?c964:239:1)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js?c964:533:1)
```

### Paste all relevant code snippets here:

Nothing due to a completely new app.

### What are detailed steps to reproduce this?

1. `blitz new blitz-new-issue`
2. Select the following items
   ```
   ✔ Pick a new project's language › TypeScript
   ✔ Pick your new app template › Full - includes DB and auth (Recommended)
   ✔ Install dependencies? › yarn
   ✔ Pick a form library (you can switch to something else later if you want) › React Final Form (recommended)
   ```
3. `cd blitz-new-issue`
4. `yarn blitz dev`
5. The following error is shown

![image](https://user-images.githubusercontent.com/436237/229410415-ceeda102-3890-48fc-acfd-fb732b2b00d4.png)

### Run `blitz -v` and paste the output here:

```
$ yarn blitz -v
yarn run v1.22.19
warning ../package.json: No license field
$ /Users/exkazuu/temp/blitz-new-issue/node_modules/.bin/blitz -v
Blitz version: 2.0.0-beta.24 (local)
macOS Ventura | darwin-arm64 | Node: v19.8.1


 Package manager: npm

  System:
    OS: macOS 13.3
    CPU: (8) arm64 Apple M1
    Memory: 305.78 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 19.8.1 - /var/folders/9p/8yqjg80d1n9dpmz33_v4hgj80000gn/T/yarn--1680495709340-0.42917398572189014/node
    Yarn: 1.22.19 - /var/folders/9p/8yqjg80d1n9dpmz33_v4hgj80000gn/T/yarn--1680495709340-0.42917398572189014/yarn
    npm: 9.5.1 - ~/.asdf/shims/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.24 => 2.0.0-beta.24
    @blitzjs/next: 2.0.0-beta.24 => 2.0.0-beta.24
    @blitzjs/rpc: 2.0.0-beta.24 => 2.0.0-beta.24
    @prisma/client: 4.6.1 => 4.6.1
    blitz: 2.0.0-beta.24 => 2.0.0-beta.24
    next: 13.2 => 13.2.4
    prisma: 4.6.1 => 4.6.1
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    typescript: ^4.8.4 => 4.9.5
```

### Please include below any other applicable logs and screenshots that show your problem:

The section of the reproduction steps includes a screenshot.

**Comment by ShaharIlany** [comment]
Same Here...

`✔ Pick a new project's language › TypeScript
✔ Pick your new app template › Full - includes DB and auth (Recommended)
✔ Install dependencies? › yarn
✔ Pick a form library (you can switch to something else later if you want) › React Hook Form`

`Blitz version: 2.0.0-beta.25 (local)
macOS Ventura | darwin-arm64 | Node: v18.15.0


 Package manager: npm

  System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Pro
    Memory: 345.95 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - /opt/homebrew/bin/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/next: 2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/rpc: 2.0.0-beta.25 => 2.0.0-beta.25 
    @prisma/client: 4.6.1 => 4.6.1 
    blitz: 2.0.0-beta.25 => 2.0.0-beta.25 
    next: 13.2 => 13.2.4 
    prisma: 4.6.1 => 4.6.1 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ^4.8.4 => 4.9.5 `

**Comment by engelkes-finstreet** [comment]
I got it for the latest beta.25 as well:

```
Blitz version: 2.0.0-beta.25 (local)
macOS Ventura | darwin-arm64 | Node: v18.14.0


 Package manager: npm

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 161.38 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.14.0 - /var/folders/by/k0t_565n2fl32vvg5lzw2jzc0000gp/T/yarn--1680559443271-0.7400009330470727/node
    Yarn: 1.22.19 - /var/folders/by/k0t_565n2fl32vvg5lzw2jzc0000gp/T/yarn--1680559443271-0.7400009330470727/yarn
    npm: 9.3.1 - ~/.nvm/versions/node/v18.14.0/bin/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/next: 2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/rpc: 2.0.0-beta.25 => 2.0.0-beta.25 
    @prisma/client: 4.6.1 => 4.6.1 
    blitz: 2.0.0-beta.25 => 2.0.0-beta.25 
    next: 13.2 => 13.2.4 
    prisma: 4.6.1 => 4.6.1 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ^4.8.4 => 4.9.5 
```

**Comment by vitaliemiron** [comment]
From my investigation error comes from the last version of next.js

change the version in your package.json to 

next: 13.2 => 13.1

It is working as a temp solution while I am investigating this

**Comment by ShaharIlany** [comment]
I tried using next 13.2(.0) but still doesn't work.

I read something about it on the beta docs of next 13 [Here](https://beta.nextjs.org/docs/api-reference/segment-config), But I am using the stable version of next13 (Pages and not App)

```
❯ npx blitz -v
Blitz version: 2.0.0-beta.25 (local)
macOS Ventura | darwin-arm64 | Node: v18.15.0


 Package manager: npm

  System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.58 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - /opt/homebrew/bin/npm
  npmPackages:
    @blitzjs/auth: ^2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/next: ^2.0.0-beta.25 => 2.0.0-beta.25 
    @blitzjs/rpc: ^2.0.0-beta.25 => 2.0.0-beta.25 
    @prisma/client: Not Found
    blitz: ^2.0.0-beta.25 => 2.0.0-beta.25 
    next: 13.2 => 13.2.4 
    prisma: Not Found
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^4.9.5 => 4.9.5 
```

**Comment by vitaliemiron** [comment]
> @vitaliemiron I tried using next 13.2(.0) but still doesn't work.
> 
> I read something about it on the beta docs of next 13 [Here](https://beta.nextjs.org/docs/api-reference/segment-config), But I am using the stable version of next13 (Pages and not App)
> 
> ```
> ❯ npx blitz -v
> Blitz version: 2.0.0-beta.25 (local)
> macOS Ventura | darwin-arm64 | Node: v18.15.0
> 
> 
>  Package manager: npm
> 
>   System:
>     OS: macOS 13.0
>     CPU: (10) arm64 Apple M1 Pro
>     Memory: 1.58 GB / 32.00 GB
>     Shell: 5.8.1 - /bin/zsh
>   Binaries:
>     Node: 18.15.0 - /usr/local/bin/node
>     Yarn: 1.22.19 - /opt/homebrew/bin/yarn
>     npm: 8.19.2 - /opt/homebrew/bin/npm
>   npmPackages:
>     @blitzjs/auth: ^2.0.0-beta.25 => 2.0.0-beta.25 
>     @blitzjs/next: ^2.0.0-beta.25 => 2.0.0-beta.25 
>     @blitzjs/rpc: ^2.0.0-beta.25 => 2.0.0-beta.25 
>     @prisma/client: Not Found
>     blitz: ^2.0.0-beta.25 => 2.0.0-beta.25 
>     next: 13.2 => 13.2.4 
>     prisma: Not Found
>     react: ^18.2.0 => 18.2.0 
>     react-dom: ^18.2.0 => 18.2.0 
>     typescript: ^4.9.5 => 4.9.5 
> ```

13.1 works for sure.

If you need 13.2 you can temporarily disable suspense inside useQuery

`export const useCurrentUser = () => {
  const [user] = useQuery(getCurrentUser, null, { suspense: false })
  return user
}`

If you have more queries, create a global query client and set suspense as default to false.

I don't understand why this error appears, may be someone from the core team could explain to us.

**Comment by IamManchanda** [comment]
I am facing the same issue 

Blitz version: 2.0.0-beta.25 (local)
macOS Ventura | darwin-arm64 | Node: v18.15.0


 Package manager: npm

  System:
    OS: macOS 13.2
    CPU: (10) arm64 Apple M1 Pro
    Memory: 160.33 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 18.15.0 - ~/.volta/tools/image/node/18.15.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 9.5.0 - ~/.volta/tools/image/node/18.15.0/bin/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.25 => 2.0.0-beta.25
    @blitzjs/next: 2.0.0-beta.25 => 2.0.0-beta.25
    @blitzjs/rpc: 2.0.0-beta.25 => 2.0.0-beta.25
    @prisma/client: 4.6.1 => 4.6.1
    blitz: 2.0.0-beta.25 => 2.0.0-beta.25
    next: 13.2 => 13.2.4
    prisma: 4.6.1 => 4.6.1
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    typescript: ^4.8.4 => 4.9.5

**Comment by filips-alpe** [comment]
Temporary fix before https://github.com/blitz-js/blitz/pull/4113 is merged:
open `package.json` and change `"next": "13.2"` to `"next": "13.2.4"` (use `npm ls next` to find your specific version)

**Comment by randomsnowflake** [comment]
I have the same issue. Changing the version number in package.json does not fix the issue for me.

**Comment by gannonh** [comment]
For those experiencing this issue, this worked for me:

- `ctrl-c`
- `rm -rf node_modules`
- `rm -rf .next`
- open `package.json` and change `"next": "13.2"` to `"next": "13.2.4"` 👆
- `npm i`
- `blitz dev`

**Comment by engelkes-finstreet** [comment]
The issue persists for me and randomly appeared again in my App. I tried all different versions:

Next 13.2.4 and 13.3.0 
Blitz 2.0.0-beta.26, 2.0.0-beta.25, 2.0.0-beta.24

Is there a way to fix this in an existing app?
The steps from @gannonh worked a few days ago but now the problem persists.

**Comment by vanam** [comment]
Same here with `"next": "^13.3.1",`

```
Blitz version: 2.0.0-beta.26 (global)
Blitz version: 2.0.0-beta.26 (local)
Linux 5.19 | linux-x64 | Node: v18.14.2


 Package manager: npm

  System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 10.15 GB / 15.44 GB
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.14.2 - ~/.nvm/versions/node/v18.14.2/bin/node
    Yarn: Not Found
    npm: 9.5.0 - ~/.nvm/versions/node/v18.14.2/bin/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.26 => 2.0.0-beta.26 
    @blitzjs/next: 2.0.0-beta.26 => 2.0.0-beta.26 
    @blitzjs/rpc: 2.0.0-beta.26 => 2.0.0-beta.26 
    @prisma/client: 4.6.1 => 4.6.1 
    blitz: 2.0.0-beta.26 => 2.0.0-beta.26 
    next: ^13.3.1 => 13.3.1 
    prisma: 4.6.1 => 4.6.1 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ^4.8.4 => 4.9.5 
```

**Comment by vanam** [comment]
@flybayer I did just `blitz new my-blitz-app` and `blitz dev` from the tutorial. It looks it attempts to apply some patch but it is not working. 

```
$ blitz dev
Loaded env from /home/martin/web2/.env.local
Loaded env from /home/martin/web2/.env
✔ Next.js was successfully patched with a React Suspense fix
✔ Routes manifest was successfully generated
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 2.3s (266 modules)
wait  - compiling...
event - compiled client and server successfully in 94 ms (266 modules)
wait  - compiling / (client and server)...
event - compiled client and server successfully in 353 ms (304 modules)
wait  - compiling...
event - compiled client and server successfully in 75 ms (304 modules)
[Rendering Suspense fallback...: DYNAMIC_SERVER_USAGE] {
  digest: 'DYNAMIC_SERVER_USAGE'
}
wait  - compiling /api/rpc/[[...blitz]] (client and server)...
event - compiled successfully in 299 ms (101 modules)
```
`blitz build` or `blitz codegen` also report patch being applied also with no success.

```
$ blitz build
Loaded env from /home/martin/web2/.env.local
Loaded env from /home/martin/web2/.env
✔ Next.js was successfully patched with a React Suspense fix
✔ Routes manifest was successfully generated
info  - Linting and checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
[    ] info  - Generating static pages (0/7)[Rendering Suspense fallback...: DYNAMIC_SERVER_USAGE] {
  digest: 'DYNAMIC_SERVER_USAGE'
}
info  - Generating static pages (7/7)
info  - Finalizing page optimization  

Route (pages)                              Size     First Load JS
┌ ○ /                                      4.78 kB         108 kB
├   └ css/73bb8e7f935a54a7.css             1.65 kB
├   /_app                                  0 B             101 kB
├ ○ /404                                   354 B           102 kB
├ λ /api/rpc/[[...blitz]]                  0 B             101 kB
├ ○ /auth/forgot-password                  1.78 kB         125 kB
├ ○ /auth/login                            1.94 kB         127 kB
├ ○ /auth/reset-password                   1.93 kB         127 kB
└ ○ /auth/signup                           1.84 kB         125 kB
+ First Load JS shared by all              101 kB
  ├ chunks/framework-2bb1844fb22814b3.js   45.2 kB
  ├ chunks/main-4df221df76d0b5c0.js        27.8 kB
  ├ chunks/pages/_app-775d35773f258c50.js  27.3 kB
  ├ chunks/webpack-0b5d8249fb15f5f3.js     939 B
  └ css/6feb208afea48685.css               240 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)

```

Same error at my Windows machine as well. I chose all default options during project generation: Typescript, Full, npm, React Final Form. Minimal version seems to work fine, though.

**Comment by LinnJS** [comment]
> @flybayer I did just `blitz new my-blitz-app` and `blitz dev` from the tutorial. It looks it attempts to apply some patch but it is not working.
> 
> ```
> $ blitz dev
> Loaded env from /home/martin/web2/.env.local
> Loaded env from /home/martin/web2/.env
> ✔ Next.js was successfully patched with a React Suspense fix
> ✔ Routes manifest was successfully generated
> ready - started server on 0.0.0.0:3000, url: http://localhost:3000
> event - compiled client and server successfully in 2.3s (266 modules)
> wait  - compiling...
> event - compiled client and server successfully in 94 ms (266 modules)
> wait  - compiling / (client and server)...
> event - compiled client and server successfully in 353 ms (304 modules)
> wait  - compiling...
> event - compiled client and server successfully in 75 ms (304 modules)
> [Rendering Suspense fallback...: DYNAMIC_SERVER_USAGE] {
>   digest: 'DYNAMIC_SERVER_USAGE'
> }
> wait  - compiling /api/rpc/[[...blitz]] (client and server)...
> event - compiled successfully in 299 ms (101 modules)
> ```
> 
> `blitz build` or `blitz codegen` also report patch being applied also with no success.
> 
> ```
> $ blitz build
> Loaded env from /home/martin/web2/.env.local
> Loaded env from /home/martin/web2/.env
> ✔ Next.js was successfully patched with a React Suspense fix
> ✔ Routes manifest was successfully generated
> info  - Linting and checking validity of types  
> info  - Creating an optimized production build  
> info  - Compiled successfully
> info  - Collecting page data  
> [    ] info  - Generating static pages (0/7)[Rendering Suspense fallback...: DYNAMIC_SERVER_USAGE] {
>   digest: 'DYNAMIC_SERVER_USAGE'
> }
> info  - Generating static pages (7/7)
> info  - Finalizing page optimization  
> 
> Route (pages)                              Size     First Load JS
> ┌ ○ /                                      4.78 kB         108 kB
> ├   └ css/73bb8e7f935a54a7.css             1.65 kB
> ├   /_app                                  0 B             101 kB
> ├ ○ /404                                   354 B           102 kB
> ├ λ /api/rpc/[[...blitz]]                  0 B             101 kB
> ├ ○ /auth/forgot-password                  1.78 kB         125 kB
> ├ ○ /auth/login                            1.94 kB         127 kB
> ├ ○ /auth/reset-password                   1.93 kB         127 kB
> └ ○ /auth/signup                           1.84 kB         125 kB
> + First Load JS shared by all              101 kB
>   ├ chunks/framework-2bb1844fb22814b3.js   45.2 kB
>   ├ chunks/main-4df221df76d0b5c0.js        27.8 kB
>   ├ chunks/pages/_app-775d35773f258c50.js  27.3 kB
>   ├ chunks/webpack-0b5d8249fb15f5f3.js     939 B
>   └ css/6feb208afea48685.css               240 B
> 
> λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
> ○  (Static)  automatically rendered as static HTML (uses no initial props)
> ```
> 
> Same error at my Windows machine as well. I chose all default options during project generation: Typescript, Full, npm, React Final Form. Minimal version seems to work fine, though.

I experienced the same when trying to create a new project this afternoon. Tried playing around with the dependancies and upgrading everything to latest and nuking the lock files and modules and still could not get a new project working.

**Comment by gjmoed** [comment]
Also ran into this, Next 13.3.1. Briefly looked into it, found this part of Blitz code-gen not matching up with actual Next source: https://github.com/blitz-js/blitz/blob/3e8f04ed03a7a4d2f8bd8d6fa6e0d00bde5b5d95/packages/blitz/src/cli/utils/codegen-tasks.ts#L46-L47

The problem here is, it's looking for `_onRecoverableError.default`, while for me, on 13.3.1 at least, it's `_onrecoverableerror.default`.

Now, manually applying the needed patch, have to watch for the same thing, towards the end of the patch, needs lowercase as well, so:
`(err) => (err.toString().includes("DYNAMIC_SERVER_USAGE") || err.toString().includes("could not finish this Suspense boundary") || err.toString().includes("Minified React error #419")) ? null : _onrecoverableerror.default(err)`

Then tossing .next directory, starting dev again, no more client errors, though server console now still does show the DYNAMIC_SERVER_USAGE suspense fallback messages.

Hope that helps anyone. Haven't looked into providing any PR for this just yet, sorry.

**Comment by gjmoed** [comment]
Looked a little bit further into this, the `_onRecoverableError.default` vs `_onrecoverableerror.default` occurs since 13.3.1. 

The onRecoverableError bit didn't appear until 13.0.7 anyways, and the function first only appears in canary for 13.0.8, which became 13.1.0.

So current patching is incorrect in various ways:
- It doesn't account for 13.0.7 at all, while the 13.0.7 code itself doesn't cover the needed changes for working with Blitz. Applying the 13 - 13.0.6 patch wouldn't work though, it'd need further work. (Blitz b26 wouldn't even run for me, with Next 13.0.x, likely wouldn't anyways, not since b24 where it started needing resolve-href from Next 13.2)

- This should be from and including 13.1.0 up to, and including 13.3.0
https://github.com/blitz-js/blitz/blob/3e8f04ed03a7a4d2f8bd8d6fa6e0d00bde5b5d95/packages/blitz/src/cli/utils/codegen-tasks.ts#L42

- And from version 13.3.1 until who knows when, we'll need the patch I suggested in my previous comment.


Here's the thing though, because of:

https://github.com/blitz-js/blitz/blob/3e8f04ed03a7a4d2f8bd8d6fa6e0d00bde5b5d95/apps/next13/package.json#L22

Why even account for anything before Next 13.3.0?

So then we should replace patching with just the 13.3.0 and >13.3.1 scenarios, or, even better, up the dependency to Next 13.3.1, and only need my suggested patch correction for 13.3.1?

@flybayer , thoughts?

**Comment by gjmoed** [comment]
Hmm, there's more trouble in paradise, found this in my Next 13.3.0 just now:

`reactRoot = _client.default.hydrateRoot(domEl, reactEl, {
    onRecoverableError:
        (err) => (err.toString().includes("DYNAMIC_SERVER_USAGE") || err.toString().includes("could not finish this Suspense boundary") || err.toString().includes("Minified React error #419")) ? null :
            (err) => (err.toString().includes("DYNAMIC_SERVER_USAGE") || err.toString().includes("could not finish this Suspense boundary") || err.toString().includes("Minified React error #419")) ? null :
                (err) => (err.toString().includes("DYNAMIC_SERVER_USAGE") || err.toString().includes("could not finish this Suspense boundary") || err.toString().includes("Minified React error #419")) ? null :
                    (err) => (err.toString().includes("DYNAMIC_SERVER_USAGE") || err.toString().includes("could not finish this Suspense boundary") || err.toString().includes("Minified React error #419")) ? null : _onRecoverableError.default(err)(err)(err)(err)
});`

So yeah, should do some checking if the patch has been applied already, or we'll keep applying again and again...

**Comment by gjmoed** [comment]
Right, so that's fairly simple:

`/_onRecoverableError\.default/` vs `/_onRecoverableError\.default$/`

**Comment by Chronosms** [comment]
Same error here with 13.3.0

Latest stable is:

`npm i --save next@^13.2.4`

Problem disappear for me

**Comment by flybayer** [comment]
We will get a fix out soon, but I wanted to share a bit of context.

Internally we're using the same pattern Next.js uses for bailing out of SSR, by [throwing an error](https://github.com/vercel/next.js/blob/97be2e672883a797ada61ebf0e1909816ea35c78/packages/next/src/shared/lib/lazy-dynamic/dynamic-no-ssr.tsx#L6C1-L10). They have the logic for [ignoring these errors on the client](https://github.com/vercel/next.js/blob/97be2e672883a797ada61ebf0e1909816ea35c78/packages/next/src/client/on-recoverable-error.ts#L18), but unfortunately [the error serialization](https://github.com/vercel/next.js/blob/2343610fdf7b196c43138f4d1acaa099f8c7ed63/packages/next/src/server/render.tsx#L384) is only implemented for the app directory, not pages directory.

So for us to be able to stop patching it, we need Nextjs to add this to pages directory.

**Comment by vanam** [comment]
Unfortunately I am still getting the same error with 2.0.0-beta.27. 😢 

I have `blitz` installed globally on linux/windows using `npm`.

```
Blitz version: 2.0.0-beta.27 (global)
Blitz version: 2.0.0-beta.27 (local)
Windows 10 | win32-x64 | Node: v18.13.0


 Package manager: npm

  System:
    OS: Windows 10 10.0.19045
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
    Memory: 6.97 GB / 31.69 GB
  Binaries:
    Node: 18.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\node_modules\yarn\bin\yarn.CMD
    npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.27 => 2.0.0-beta.27
    @blitzjs/next: 2.0.0-beta.27 => 2.0.0-beta.27
    @blitzjs/rpc: 2.0.0-beta.27 => 2.0.0-beta.27
    @prisma/client: 4.6.1 => 4.6.1
    blitz: 2.0.0-beta.27 => 2.0.0-beta.27
    next: ^13.3.0 => 13.3.4
    prisma: 4.6.1 => 4.6.1
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    typescript: ^4.8.4 => 4.9.5

```

**Comment by muco-rolle** [comment]
The same for me a fresh install of beta 2.0.0-beta.27 and I still get the same error

**Comment by elsueno** [comment]
This ticket should be reopened because the problem still exists with blitz beta 27 when running the commands below.

```
blitz new myAppName
cd myAppName
blitz dev
```

```Blitz version: 2.0.0-beta.27 (global)
Blitz version: 2.0.0-beta.27 (local)
Linux 5.19 | linux-x64 | Node: v18.10.0


 Package manager: npm

  System:
    OS: Linux 5.19 Pop!_OS 22.04 LTS
    CPU: (16) x64 AMD Ryzen 7 PRO 4750U with Radeon Graphics
    Memory: 3.33 GB / 14.86 GB
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.10.0 - ~/.nvs/node/18.10.0/x64/bin/node
    Yarn: 1.22.19 - ~/.nvs/node/18.10.0/x64/bin/yarn
    npm: 8.19.2 - ~/.nvs/node/18.10.0/x64/bin/npm
  npmPackages:
    @blitzjs/auth: 2.0.0-beta.27 => 2.0.0-beta.27 
    @blitzjs/next: 2.0.0-beta.27 => 2.0.0-beta.27 
    @blitzjs/rpc: 2.0.0-beta.27 => 2.0.0-beta.27 
    @prisma/client: 4.6.1 => 4.6.1 
    blitz: 2.0.0-beta.27 => 2.0.0-beta.27 
    next: ^13.3.0 => 13.4.3 
    prisma: 4.6.1 => 4.6.1 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ^4.8.4 => 4.9.5 
```

**Comment by bravo-kernel** [comment]
The issue is real, below solution as well:

1. Delete folder `node_modules`
2. **Delete file `package-lock.json`**
3. Delete folder `.next`
4. Re-run `npm install`

**Comment by KPouianou** [comment]
I have the exact same error with:
`"next": "13.4.4"`
`"blitz": "2.0.0-beta.27"`

I am only seeing this in dev mode with `blitz dev`. in production, `blitz build && blitz start` seems to work.

**Comment by bravo-kernel** [comment]
Probably unrelated but I just checked the commit that fixed the dev issue for me and noticed that I also bumped all next packages to `13.4.4`. Basically as the first step, before running the other five described above.

```
    "next": "13.4.4",
    "@next/bundle-analyzer": "13.4.4",
    "eslint-config-next": "13.4.4",
```

**Comment by exKAZUu** [comment]
I can still reproduce the issue, so I've created a new issue with re-created reproduction repo.

**Comment by tordans** [comment]
Using @bravo-kernel's steps combined with the update to next 13.4.4 does work for me.

```shell
npx blitz@latest new # TS, +DB, npm, React Hook Form
npm i next@13.3.4
npx blitz@latest start
```

**Comment by exKAZUu** [comment]
This issue should be fixed via #4150.
